scipy/stats/_hypotests.py

```diff
@@ -377,15 +376,22 @@ def cramervonmises(rvs, cdf, args=()):
 
 def _get_wilcoxon_distr(n):
     """
-    Distribution of counts of the Wilcoxon ranksum statistic r_plus (sum of
+    Distribution of probability of the Wilcoxon ranksum statistic r_plus (sum of
     ranks of positive differences).
-    Returns an array with the counts/frequencies of all the possible ranks
+    Returns an array with the probabilities of all the possible ranks
     r = 0, ..., n*(n+1)/2
     """
-    cnt = _wilcoxon_data.COUNTS.get(n)
-
-    if cnt is None:
-        raise ValueError("The exact distribution of the Wilcoxon test "
-                         "statistic is not implemented for n={}".format(n))
-
-    return np.array(cnt, dtype=int)
+    result = []
+    c = np.ones(1, dtype=np.double)
+    result.append(c)
+
+    # uint64max = 2**64 - 1
+    for k in range(1, n + 1):
+        prev_c = c
+        c = np.zeros(k * (k + 1) // 2 + 1, dtype=np.double)
+        m = len(prev_c)
+        c[:m] = prev_c * 0.5
+        c[-m:] += prev_c * 0.5
+        result.append(c)
+
+    return result[n]
```

Comment:
If only the nth is returned, why store the rest?
We could hold on to all of `result` in case they can be re-used in another call to the function. If it's not worth the trouble, maybe we don't need to form the list.

Reply:
```suggestion
    c = np.ones(1, dtype=np.double)
    for k in range(1, n + 1):
        prev_c = c
        c = np.zeros(k * (k + 1) // 2 + 1, dtype=np.double)
        m = len(prev_c)
        c[:m] = prev_c * 0.5
        c[-m:] += prev_c * 0.5
        
    return c
```
You are right, there is no need, I used "result" to compare to the previous lookup table/ implementation